electrum/plugins/keepkey/qt.py

```diff
@@ -272,30 +285,29 @@ def set_enabled():
             hbox_pin.addStretch(1)
 
         if method in [TIM_NEW, TIM_RECOVER]:
-            vbox.addWidget(WWLabel(RECOMMEND_PIN))
-            vbox.addWidget(cb_pin)
+            self.addWidget(WWLabel(RECOMMEND_PIN))
+            self.addWidget(self.cb_pin)
         else:
-            vbox.addLayout(hbox_pin)
+            self.addLayout(hbox_pin)
 
         passphrase_msg = WWLabel(PASSPHRASE_HELP_SHORT)
         passphrase_warning = WWLabel(PASSPHRASE_NOT_PIN)
         passphrase_warning.setStyleSheet("color: red")
-        cb_phrase = QCheckBox(_('Enable passphrases'))
-        cb_phrase.setChecked(False)
-        vbox.addWidget(passphrase_msg)
-        vbox.addWidget(passphrase_warning)
-        vbox.addWidget(cb_phrase)
-
-        wizard.exec_layout(vbox, next_enabled=next_enabled)
-
-        if method in [TIM_NEW, TIM_RECOVER]:
-            item = bg.checkedId()
-            pin = cb_pin.isChecked()
+        self.cb_phrase = QCheckBox(_('Enable passphrases'))
+        self.cb_phrase.setChecked(False)
+        self.addWidget(passphrase_msg)
+        self.addWidget(passphrase_warning)
+        self.addWidget(self.cb_phrase)
+
+    def get_settings(self):
+        if self.method in [TIM_NEW, TIM_RECOVER]:
+            item = self.bg.checkedId()
+            pin = self.cb_pin.isChecked()
         else:
-            item = ' '.join(str(clean_text(text)).split())
-            pin = str(pin.text())
+            item = ' '.join(str(clean_text(self.text_e)).split())
+            pin = str(self.pin.text())
```

Comment:
Testing with an uninitialized keepkey. When using `TIM_MNEMONIC`  or `TIM_PRIVKEY` to restore, getting:
```
Traceback (most recent call last):
  File "...\electrum\electrum\gui\qt\wizard\wizard.py", line 151, in on_page_updated
	page.apply()
  File "...\electrum\electrum\plugins\keepkey\qt.py", line 656, in apply
	self.wizard_data['keepkey_settings'] = self.settings_layout.get_settings()
  File "...\electrum\electrum\plugins\keepkey\qt.py", line 308, in get_settings
	pin = str(self.pin.text())
AttributeError: 'KeepkeyInitLayout' object has no attribute 'pin'
```

Reply:
fixed in ec4c2487b810cfe8a13b709fc399458f1d29cbce